File: src_a_(C++)/Test5_12a.c
//Ïğîåêò Ñòğàííèê Ìîäóëà-Ñè-Ïàñêàëü Äëÿ Windows 32, òåñòîâàÿ ïğîãğàììà
//Ãğóïïà òåñòîâ 5:ÏĞÎÖÅÄÓĞÛ
//Òåñò íîìåğ    12:ÂÛÇÎÂ ÈÇ ÂÛØÅËÅÆÀÙÅÃÎ ÌÎÄÓËß

include Win32

char s[15];

int i;

void pr1(int j);
void pr2(int j);

void pr1(int j)
{
  if(j>0) pr2(j-1);
  else return;
  i++;
}

File: src_a_(Modula)/Test5_12a.m
//Ïğîåêò Ñòğàííèê-Ìîäóëà Äëÿ Windows 32, òåñòîâàÿ ïğîãğàììà
//Ãğóïïà òåñòîâ 5:ÏĞÎÖÅÄÓĞÛ
//Òåñò íîìåğ    12:ÂÛÇÎÂ ÈÇ ÂÛØÅËÅÆÀÙÅÃÎ ÌÎÄÓËß
implementation module Test5_12a;
import Win32;

var s:string[15];

var i:integer;

procedure pr1(j:integer); forward; 
procedure pr2(j:integer); forward;

procedure pr1;
begin
  if j>0
    then pr2(j-1)
    else return
  end;
  i:=i+1
end pr1;

end Test5_12a.

